middleware/fstype_test.go

```diff
@@ -0,0 +1,24 @@
+package middleware
+
+import (
+	"runtime"
+	"testing"
+)
+
+func TestCaseSensitiveFS(t *testing.T) {
+	caseSensitive := false
+	switch runtime.GOOS {
+	case "linux":
+		caseSensitive = true
+	case "darwin":
+		caseSensitive = false
```

Comment:
Alright. Let's do that. It'll be interesting to see how long it takes before someone runs the tests on a filesystem other than their system's default ;)